### PR TITLE
SPARK-11099 [SPARK SHELL] [SPARK SUBMIT] Default conf property file i…

### PR DESCRIPTION
**Pull request by zjffdu**
Please help review it. Thanks

### CODE REVIEW

**Comment by srowen** [comment]
CC @vanzin I'm no expert but I do see the point that this seems to be checking "is there a non-default props file set" only.

File: launcher/src/test/java/org/apache/spark/launcher/SparkSubmitCommandBuilderSuite.java [hunk 1]
```diff
@@ -164,11 +166,16 @@ private void testCmdBuilder(boolean isDriver) throws Exception {
...
-    launcher.conf.put(SparkLauncher.DRIVER_EXTRA_LIBRARY_PATH, "/native");
     launcher.conf.put("spark.foo", "foo");
+    // either set the property through "--conf" or through property file
+    if (!usePropertyFile) {
+      launcher.conf.put(SparkLauncher.DRIVER_MEMORY, "1g");
+      launcher.conf.put(SparkLauncher.DRIVER_EXTRA_CLASSPATH, "/driver");
+      launcher.conf.put(SparkLauncher.DRIVER_EXTRA_JAVA_OPTIONS, "-Ddriver -XX:MaxPermSize=256m");
+      launcher.conf.put(SparkLauncher.DRIVER_EXTRA_LIBRARY_PATH, "/native");
+    } else {
+      launcher.childEnv.put("SPARK_CONF_DIR", "test/resources");
```
Comment (vanzin) [inline launcher/src/test/java/org/apache/spark/launcher/SparkSubmitCommandBuilderSuite.java]: I'm not sure this works the way you expect it too, and is probably the cause of the test failure. You'll probably need to build the full path based on `System.getProperty("spark.test.home")` (which is the root of the repo).

**Comment by zjffdu** [comment]
@vanzin Failure unit test is fixed, please help review.

File: launcher/src/test/java/org/apache/spark/launcher/SparkSubmitCommandBuilderSuite.java [hunk 2]
```diff
@@ -80,7 +82,7 @@ public void testCliParser() throws Exception {
     assertTrue("Driver -Xms should be configured.", cmd.contains("-Xms42g"));
     assertTrue("Driver -Xmx should be configured.", cmd.contains("-Xmx42g"));
     assertTrue("Command should contain user-defined conf.",
-      Collections.indexOfSubList(cmd, Arrays.asList(parser.CONF, "spark.randomOption=foo")) > 0);
+        Collections.indexOfSubList(cmd, Arrays.asList(parser.CONF, "spark.randomOption=foo")) > 0);
```
Comment (vanzin) [inline launcher/src/test/java/org/apache/spark/launcher/SparkSubmitCommandBuilderSuite.java]: nit: previous indentation was correct. we don't double-indent in Spark (aside from rare exceptions).

**Comment by vanzin** [comment]
LGTM, I'll fix the nit while merging.

**Pull request closed without merging**